amethyst_gltf/src/lib.rs

```diff
@@ -6,141 +6,228 @@ extern crate base64;
+    }
 }
 
-/// A single scene is defined as a list of the root nodes in the node hierarchy for the full asset
-#[derive(Debug)]
-pub struct GltfScene {
-    pub root_nodes: Vec<usize>,
+impl From<Range<[f32; 3]>> for GltfNodeExtent {
+    fn from(range: Range<[f32; 3]>) -> Self {
+        GltfNodeExtent {
+            start: Point3::from(range.start),
+            end: Point3::from(range.end),
+        }
+    }
 }
 
-/// A single animation
-#[derive(Debug)]
-pub struct GltfAnimation {
-    // node index, vec will be same size as samplers and channels, and reference the sampler+channel
-    // at the same index
-    pub samplers: Vec<(usize, TransformChannel, Sampler<SamplerPrimitive<f32>>)>,
-    pub handle: Option<Handle<Animation<Transform>>>,
-    //pub hierarchy_root: usize,
+impl Component for GltfNodeExtent {
+    type Storage = DenseVecStorage<Self>;
 }
 
 /// Options used when loading a GLTF file
-#[derive(Debug, Clone, Default)]
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(default)]
 pub struct GltfSceneOptions {
-    pub generate_tex_coords: Option<(f32, f32)>,
+    /// Generate texture coordinates if none exist in the Gltf file
+    pub generate_tex_coords: (f32, f32),
+    /// Load animation data from the Gltf file
     pub load_animations: bool,
+    /// Flip the v coordinate for all texture coordinates
     pub flip_v_coord: bool,
-    pub move_to_origin: bool,
-}
-
-/// Actual asset produced on finished loading of a GLTF scene file.
-#[derive(Debug)]
-pub struct GltfSceneAsset {
-    pub nodes: Vec<GltfNode>,
-    pub scenes: Vec<GltfScene>,
-    pub materials: Vec<GltfMaterial>,
-    pub animations: Vec<GltfAnimation>,
-    pub default_scene: Option<usize>,
-    pub skins: Vec<GltfSkin>,
-    pub options: GltfSceneOptions,
+    /// Load the given scene index, if not supplied will either load the default scene (if set),
+    /// or the first scene (only if there is only one scene, otherwise an `Error` will be returned).
+    pub scene_index: Option<usize>,
 }
 
-impl Into<Result<ProcessingState<GltfSceneAsset>, AssetError>> for GltfSceneAsset {
-    fn into(self) -> Result<ProcessingState<GltfSceneAsset>, AssetError> {
-        Ok(ProcessingState::Loaded(self))
+impl<'a> PrefabData<'a> for GltfPrefab {
+    type SystemData = (
+        <Transform as PrefabData<'a>>::SystemData,
+        <MeshData as PrefabData<'a>>::SystemData,
+        <MaterialPrefab<TextureFormat> as PrefabData<'a>>::SystemData,
+        <AnimatablePrefab<usize, Transform> as PrefabData<'a>>::SystemData,
+        <SkinnablePrefab as PrefabData<'a>>::SystemData,
+        WriteStorage<'a, GltfNodeExtent>,
+    );
+    type Result = ();
+
+    fn load_prefab(
+        &self,
+        entity: Entity,
+        system_data: &mut Self::SystemData,
+        entities: &[Entity],
+    ) -> Result<(), Error> {
+        if let Some(ref transform) = self.transform {
+            transform.load_prefab(entity, &mut system_data.0, entities)?;
```

Comment:
Tuple is getting pretty large and should be destructured into more friendly names.